Test/ServerTelemetryChannel.Test/Shared.Tests/Implementation/TelemetryProcessorChainBuilderExtensionsTest.cs

```diff
@@ -0,0 +1,172 @@
+            var channelBuilder = new TelemetryProcessorChainBuilder(tc);
+
+            Assert.Throws<ArgumentNullException>(() => channelBuilder.UseAdaptiveSampling(null, null));
+        }
+
+        [TestMethod]
+        public void UseAdaptiveSamplingAddsAdaptiveSamplingProcessorToTheChainWithCorrectSettings()
+        {
+            SamplingPercentageEstimatorSettings settings = new SamplingPercentageEstimatorSettings
+            {
+                MaxSamplingPercentage = 13
+            };
+            AdaptiveSamplingPercentageEvaluatedCallback callback = (second, percentage, samplingPercentage, changed, estimatorSettings) => { };
+
+            var tc = new TelemetryConfiguration { TelemetryChannel = new StubTelemetryChannel() };
+            var channelBuilder = new TelemetryProcessorChainBuilder(tc);
+            channelBuilder.UseAdaptiveSampling(settings, callback);
+            channelBuilder.Build();
+
+            Assert.Equal(13, ((AdaptiveSamplingTelemetryProcessor)tc.TelemetryProcessorChain.FirstTelemetryProcessor).MaxSamplingPercentage);
+        }
+
+        [TestMethod]
+        public void UseAdaptiveSamplingWithSettingsParameterAndExcludedTypesThrowsArgumentNullExceptionBuilderIsNull()
+        {
+            Assert.Throws<ArgumentNullException>(() => TelemetryProcessorChainBuilderExtensions.UseAdaptiveSampling(null, null, null, "request"));
+        }
+
+        [TestMethod]
+        public void UseAdaptiveSamplingWithSettingsParameterAndExcludedTypesThrowsArgumentNullExceptionSettingsIsNull()
+        {
+            var tc = new TelemetryConfiguration { TelemetryChannel = new StubTelemetryChannel() };
+            var channelBuilder = new TelemetryProcessorChainBuilder(tc);
+
+            Assert.Throws<ArgumentNullException>(() => channelBuilder.UseAdaptiveSampling(null, null, "request"));
+        }
+
+        [TestMethod]
+        public void UseAdaptiveSamplingAddsAdaptiveSamplingProcessorToTheChainWithCorrectSettingsAndExcludedTypes()
+        {
+            SamplingPercentageEstimatorSettings settings = new SamplingPercentageEstimatorSettings
+            {
+                MaxSamplingPercentage = 13
+            };
+            AdaptiveSamplingPercentageEvaluatedCallback callback = (second, percentage, samplingPercentage, changed, estimatorSettings) => { };
+
+            var tc = new TelemetryConfiguration { TelemetryChannel = new StubTelemetryChannel() };
+            var channelBuilder = new TelemetryProcessorChainBuilder(tc);
+            channelBuilder.UseAdaptiveSampling(settings, callback, "request");
+            channelBuilder.Build();
+
+            Assert.Equal(13, ((AdaptiveSamplingTelemetryProcessor)tc.TelemetryProcessorChain.FirstTelemetryProcessor).MaxSamplingPercentage);
+            Assert.Equal("request", ((AdaptiveSamplingTelemetryProcessor)tc.TelemetryProcessorChain.FirstTelemetryProcessor).ExcludedTypes);
+        }
```

Comment:
They test different extension methods

---
In reply to: [67446623](https://github.com/Microsoft/ApplicationInsights-dotnet/pull/243#discussion_r67446623) [](ancestors = 67446623)